sys/dev_vga.c

```diff
@@ -102,10 +103,43 @@ static int dev_vga_lookup(vnode_t *v, const char *name, vnode_t **res) {
   return ENOENT;
 }
 
+static void uiomove_vga_dirent(const char *name, uio_t *uio) {
+  dirent_t *dir = NULL;
+  dir = (dirent_t *)kmalloc(M_VFS, _DIRENT_RECLEN(dir, strlen(name)), M_ZERO);
+  dir->d_namlen = strlen(name);
+  dir->d_reclen = _DIRENT_RECLEN(dir, dir->d_namlen);
+  memcpy(dir->d_name, name, dir->d_namlen + 1);
+  uiomove(dir, dir->d_reclen, uio);
+  kfree(M_VFS, dir);
+}
+
```

Comment:
Please declare following array here:
```c
typedef struct direntinfo {
  uint8_t d_type;
  char *d_name;
} direntinfo_t;

static direntinfo_t vga_dir[] = {
  {DT_BLK, "fb"},
  {DT_BLK, "palette"},
  {DT_CHR, "videomode"},
};
```
... and use it to create a memory block that consists of `dirent`s.